mitodl/teachersportal
# Added fields to Course, change course API to not contact CCXCon anymore

#### What are the relevant tickets?

Fixes #437 
Fixes #425 
#### What's this PR do?
- Adds fields to Course which are populated by new fields in the webhooks
- Removes code to contact CCXCon in the course API
- Updates the front end to account for the moved fields
- Adds `CourseSerializer` and `ModuleSerializer` to replace `course_as_dict` and `module_as_dict`
#### Where should the reviewer start?

portal/models.py
#### How should this be manually tested?

First, make sure you have the latest master from CCXCon. Adjust the Advanced Settings in your devstack to cause the webhook to execute. Then view the course listing and course detail page for a course. The course listing and course detail page should look exactly the same as before. You should test this while logged in and logged out.
#### Background context

Some of the validation done for course PATCH is unnecessary now but I'd like to push that off until another PR sometime in the future to prevent this one from blowing up.
#### What GIF best describes this PR or how it makes you feel?

![](http://i.imgur.com/v5peAgO.gif)
